hibernate-core/src/test/java/org/hibernate/test/rowid/RowIdTest.java

```diff
@@ -42,50 +56,58 @@ public void afterSessionFactoryBuilt() {
 		super.afterSessionFactoryBuilt();
 		final Session session = sessionFactory().openSession();
 		session.doWork(
-				new Work() {
-					@Override
-					public void execute(Connection connection) throws SQLException {
-						Statement st = ((SessionImplementor)session).getJdbcCoordinator().getStatementPreparer().createStatement();
-						try {
-							((SessionImplementor)session).getJdbcCoordinator().getResultSetReturn().execute( st, "drop table Point");
-						}
-						catch (Exception ignored) {
-						}
-						((SessionImplementor)session).getJdbcCoordinator().getResultSetReturn().execute( st, "create table Point (\"x\" number(19,2) not null, \"y\" number(19,2) not null, description varchar2(255) )");
-						((SessionImplementor)session).getJdbcCoordinator().getResourceRegistry().release( st );
+				connection -> {
+					Statement st = ((SessionImplementor)session).getJdbcCoordinator().getStatementPreparer().createStatement();
+					try {
+						((SessionImplementor)session).getJdbcCoordinator().getResultSetReturn().execute( st, "drop table Point");
+					}
+					catch (Exception ignored) {
 					}
+					((SessionImplementor)session).getJdbcCoordinator().getResultSetReturn().execute( st, "create table Point (\"x\" number(19,2) not null, \"y\" number(19,2) not null, description varchar2(255) )");
+					((SessionImplementor)session).getJdbcCoordinator().getResourceRegistry().release( st );
 				}
 		);
 		session.close();
 	}
 
 	@Test
-	public void testRowId() {
-		Session s = openSession();
-		Transaction t = s.beginTransaction();
-		Point p = new Point( new BigDecimal(1.0), new BigDecimal(1.0) );
-		s.persist(p);
-		t.commit();
-		s.clear();
-		
-		t = s.beginTransaction();
-		p = (Point) s.createCriteria(Point.class).uniqueResult();
-		p.setDescription("new desc");
-		t.commit();
-		s.clear();
-		
-		t = s.beginTransaction();
-		p = (Point) s.createQuery("from Point").uniqueResult();
-		p.setDescription("new new desc");
-		t.commit();
-		s.clear();
-		
-		t = s.beginTransaction();
-		p = (Point) s.get(Point.class, p);
-		p.setDescription("new new new desc");
-		t.commit();
-		s.close();
+	public void testUpdateByRowId() {
+		doInHibernate( this::sessionFactory, s -> {
+			Point p = new Point( new BigDecimal(1.0), new BigDecimal(1.0) );
```

Comment:
minor : Idea higlights it as "Unpredictable 'new BigDecimal()' call"
![image](https://user-images.githubusercontent.com/6348215/77051217-63708580-69e4-11ea-8cbb-bf7681c70b19.png)

```suggestion
			Point p = new Point( new BigDecimal("1.0"), new BigDecimal("1.0") );
```

Reply:
I think that `BigDecimal.valueOf` might be a good alternative.

Reply:
Yep, I didn't know about that

```suggestion
			Point p = new Point( BigDecimal.valueOf(1.0), BigDecimal.valueOf(1.0) );
```